File: 2017-10-30-338-easy-what-day-was-it-again/main.cpp
#include <iostream>
#include <vector>

static const std::vector<std::string> weekdays = { "Saturday", "Sunday", "Monday",
	"Tuesday", "Wednesday", "Thursday", "Friday" };

int main()
{
	int year, month, day;
	while (std::cin >> year >> month >> day) {
		if (month < 3) {
			month += 12;
			year--;
		}
		int k = year % 100, j = year / 100;
		int result = (day + (13 * (month + 1) / 5) + k + (k / 4) + (j / 4) + (5 * j)) % 7;
		std::cout << weekdays.at(result) << '\n';
	}
	return 0;
}

File: 2017-10-30-338-easy-what-day-was-it-again/main.swift
import Foundation

let weekdays = ["Saturday", "Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday"]

func calculateDay(forYear year: Int, forMonth month: Int, forDay day: Int) -> String {
    var m = month, y = year
    if m < 3 {
        m += 12
        y -= 1
    }
    let k = y % 100, j = y / 100
    let index = (day + (13 * (m + 1) / 5) + k + (k / 4) + (j / 4) + (5 * j)) % 7
    return weekdays[index]
}